static/js/publisher/form/mediaList.js

```diff
@@ -26,7 +26,14 @@ class MediaList extends React.Component {
         classes.push("u-hide");
       }
 
-      inputs.push(
+      // We always want the "add" item to be before any hidden inputs
```

Comment:
This is because of the CSS magic I do later with nth-child... The "Add" element was always added to the end so wasn't being caught by the selector.

Reply:
Is this still relevant if we don't use grid for media images?

Reply:
it's not needed, but I think it's nice to have - to make visible items consistently together

Reply:
OK. I don't fully understand what it does, so let's stick to it ;)

Reply:
Ah - just so you know, the "Add screenshot" was always getting appended to the list of inputs. So you've have - "visible, visible, visible, hidden, hidden, visible" order of elements. This just makes sure it's "visible, visisble, visible, visible, hidden, hidden"... I'm sure that explains it completely :|